### PR TITLE
BS-13900 Repository initialization

### PR DESCRIPTION
**Pull request by rbioteau**
* Check for .classpath and MANIFEST when opening a repository (those
files are not committed on the server)
* Use a proper design to create a repository instance in order to make
it testable (with di)

https://github.com/bonitasoft/bonita-studio-sp/pull/58

### CODE REVIEW

File: bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/core/ProjectManifestFactory.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    protected Map<String, String> createManifestHeaders(final String projectName) {
+        final Map<String, String> headers = new HashMap<String, String>();
+        headers.put(ManifestUtils.MANIFEST_VERSION, "1.0");
+        headers.put(org.osgi.framework.Constants.BUNDLE_MANIFESTVERSION, "2");
+        headers.put(org.osgi.framework.Constants.BUNDLE_NAME, projectName);
+        headers.put(org.osgi.framework.Constants.BUNDLE_SYMBOLICNAME, toJavaIdentifier(projectName, false));
+        headers.put(org.osgi.framework.Constants.BUNDLE_VERSION, "1.0.0.qualifier");
+        headers.put(org.osgi.framework.Constants.BUNDLE_VENDOR, "BonitaSoft S.A.");
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/core/ProjectManifestFactory.java]: we can take th eopportunity to uodate the BonitaSoft --> Bonitasoft

File: bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/core/ProjectManifestFactory.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        headers.put(org.osgi.framework.Constants.BUNDLE_REQUIREDEXECUTIONENVIRONMENT, "JavaSE-1.7");
+        headers.put(org.osgi.framework.Constants.REQUIRE_BUNDLE,
+                new StringBuilder()
+                        .append("javax.persistence;bundle-version=\"2.0.3\"").append(",")
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/core/ProjectManifestFactory.java]: we have an hardcoded version on javax.persistence? Maybe we can take a look to get rid of that.

File: bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/extension/ExtensionContextInjectionFactory.java [hunk 3]
```diff
@@ -50,6 +50,13 @@ public static IEclipseContext workbenchContext() {
             }
             throw new IllegalStateException("No workbench available");
         }
+        return headlessContext();
+    }
+
+    private static IEclipseContext headlessContext() {
```
Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common/src/org/bonitasoft/studio/common/extension/ExtensionContextInjectionFactory.java]: shouldn't it be synchronized? (SIngleton)

File: bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/core/ProjectManifestFactory.java (hunk 2 above)
Comment (rbioteau) [inline bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/core/ProjectManifestFactory.java]: I don't if's a good idea (perf will be impacted)
and osgi version are not strict so if we update to a newer version it should not be impacted

Comment (apupier) [inline bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/core/ProjectManifestFactory.java]: OK

**Pull request merged**